packages/deployments/contracts/src/init/initProtocol.ts

```diff
@@ -0,0 +1,122 @@
+      // Address of the bridged asset on this domain.
+      local: string;
+      // Address of the adopted asset on this domain.
+      // NOTE: If adopted is specified, a stableswap will be initialized! If not
+      // specified, then we assume the local asset is the adopted asset on this domain.
+      adopted?: string;
+    };
+  };
+};
+
+type DomainStack = {
+  // Meta info.
+  chain: number;
+  domain: string;
+
+  // RPC provider to use for this network.
+  rpc: string;
+
+  deployments: {
+    // Diamond.
+    Connext: string;
+    // Handlers.
+    BridgeRouter: string;
+    RelayerFeeRouter: string;
+    PromiseRouter: string;
+    // Registry.
+    TokenRegistry: string;
+
+    // Messaging Layer.
+    // ConnectorManager
+    // SendOutboundRootResolver
+
+    // The messaging layer deployments are different depending on whether this
+    // is the hub domain or spoke domain.
+    messaging:
+      | {
+          RootManager: string;
+          MainnetConnector: string;
+          HubConnectors: string;
+        }
+      | {
+          SpokeConnector: string;
+        };
+  };
+};
+
+type ProtocolStack = {
+  hub: string; // The hub domain.
+  // Domain stack should have all info pertaining to each supported domain.
+  domains: DomainStack[];
+  // Crosschain ERC20 assets to enroll in TokenRegistry.
+  assets: AssetStack[];
+  // Agents that need to be whitelisted (across all domains).
+  // Leave undefined if no agents should be whitelisted in this setup.
+  agents?: WhitelistAgents;
+};
+
+/**
+ * Handle configuration of the entire protocol, including messaging stack and connext diamond
+ * contracts, across all listed domains.
+ *
+ * Should effectively be a diagnostic on the whole protocol, making sure that everything that
+ * requires configuration and/or setup has been done so properly.
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const configureProtocol = async (protocol: ProtocolStack) => {
+  /// ********************** SETUP **********************
+  /// MARK - Sanity Checks
+  // Hub domain should be included in domains.
+  // All domains specified in AssetStack(s) must be included in domains.
+  // DomainStack for hub includes proper messaging deployments.
+  // All spoke domains should have proper messaging deployments.
+  /// MARK - Deployer
+  // Get deployer mnemonic, which should be provided in env.
+  /// MARK - Deployments
+  // Get all deployments
+  /// ******************** MESSAGING ********************
+  /// MARK - Init
+  // TODO: Currently unused, as messaging init checks are not needed with the AMB-compatible stack.
+  // However, they will be useful as sanity checks for Nomad deployments in the future - thus, leaving
+  // this placeholder here for now...
+  /// MARK - Connector Mirrors
+  // Connectors should have their mirrors' address set; this lets them know about their counterparts.
+  /// MARK - Enroll Handlers
+  // Whitelist messaging routers as callers of dispatch?
```

Comment:
yep! `addSender` on connector -- this may be done in the deployment already but you will need to make sure all router (message sending) contracts are whitelisted on the connector